new-frontiers-14/frontier-station-14
# Turrets: updated sprites and new entities

## About the PR
### Reworked disabler turrets added in [this PR](https://github.com/new-frontiers-14/frontier-station-14/pull/1594):
> A Contraband Denying Energy Turret (CDET). A disabler turret with built-in microreactor. Use screwdriver to disassemble.
- New sprites made by Stagnation (discord).
- Reworked NT disabler turrets can be deployed and packed.

### Reworked laser turrets added in [this PR](https://github.com/new-frontiers-14/frontier-station-14/pull/1594):
> A Self-Recharging Laser Auto Turret (SLAT). A laser turret with built-in microreactor. Use screwdriver to disassemble.
- New sprites made by Stagnation (discord).
- Reworked NT laser turrets can be deployed and packed.

### Added new ballistic turret:
> An Autonomous Sentry Machine Gun Turret (ASMGT) is a magazine fed ballistic turret compatible with .20, .25, .30 and .35 cartriges. Use screwdriver to disassemble.
- Sprites for turret and ammo boxes made by Stagnation (discord).
- Magazine fed.
- Can be deployed and packed.

### Other changes:
- Added heavy weapon case for new packed turrets.
- Added SR gun safe, filled with mk58, 2 pistol mags and heavy weapon case with two Contraband Denying Energy Turret. Replaced custom filled safe in SR office with new safe.
- Added NFSD gun safe, filled with two heavy weapon cases with Contraband Denying Energy Turrets and one heavy weapon case with Self-Recharging Laser Auto Turret. Replaced empty safe in armory with new safe.
- @whatston3 made c# code that allows disabler turret ignore stunned characters.

## To Do:
- [x] In hand sprites for turret kits.
- [x] In hand sprites for heavy weapon case.
- [x] Expedition loot tables integration for some turrets.
- [x] Map contraband denying turret weapon cases to NFSD outpost and SR office.

## Why / Balance
New tools for NFSD/SR and players.

## How to test
1. Spawn `WeaponCaseHeavyAsmgt`
2. Use 'packed ASMGT' to deploy turret (unpacking works like inflatable walls - take the item in your hands, click with it on unoccupied tile).
3. Put ammo mag from the weapon case to arm the turret.
4. Spawn hostile entities.
5. To "pack" turret back use screwdriver on it.
6. Spawn `WeaponCaseHeavyCdet` and `WeaponCaseHeavySlat`
7. Use 'packed ...' to deploy turrets.
8. Spawn hostile entities.
9. Spawn and wear contraband gear while close to CDET - once you put on contraband clothes the turret should become hostile towards your character, once you remove contraband clothes the turret should become neutral again.
10. To "pack" turret back use screwdriver on it.
11. Spawn `WeaponTurretAsmgtHostileUniversallyPacked`
12. Upon deployment should be friendly towards your character, but hostile to anything else. Same deal as with killer tomatoes.

## Media

![2024-7-18_17 35 08](https://github.com/user-attachments/assets/d9baeef6-ee33-47cb-8b58-e1f44d70a994)

https://github.com/user-attachments/assets/9c38adb4-cc17-4f3c-a926-c7dd6aadac38

https://github.com/user-attachments/assets/8e017e8a-21be-48fb-a5ec-b39507812850

https://github.com/user-attachments/assets/13bd8f54-3ca9-42ce-8346-9132a3b595c9


## Breaking changes

**Changelog**
:cl: erhardsteinhauer
- add: Added magazine fed ballistic turret. Obtainable from expeditions.
- add: Added heavy weapon cases with turrets to SR's gun safe and to NFSD armory.
